com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs

```diff
@@ -626,6 +654,154 @@ static void OnLightingDataCleared()
+            using var validity = new NativeArray<float>(asset.totalCellCounts.probesCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+            using var offsets = new NativeArray<Vector3>(asset.totalCellCounts.offsetsCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+
+            var startCounts = new ProbeVolumeAsset.CellCounts();
+            for (var i = 0; i < bakingCells.Count; ++i)
+            {
+                var bakingCell = bakingCells[i];
+                var cellCounts = asset.cellCounts[i];
+
+                bricks.GetSubArray(startCounts.bricksCount, cellCounts.bricksCount).CopyFrom(bakingCell.bricks);
+
+                var coefficientVectorsCount = asset.CoefficientVectorsCount;
+                var probesTarget = probes.GetSubArray(startCounts.probesCount * coefficientVectorsCount, cellCounts.probesCount * coefficientVectorsCount);
+                for (int j = 0, k = 0; j < cellCounts.probesCount; ++j, k += coefficientVectorsCount)
+                {
+                    ref var sh = ref bakingCell.sh[j];
+
+                    GetShaderCoeffsL0L1(ref sh, probesTarget, k);
+
+                    if (asset.bands == ProbeVolumeSHBands.SphericalHarmonicsL2)
+                        GetShaderCoeffsL2(ref sh, probesTarget, k + 3);
+                }
+
+                positions.GetSubArray(startCounts.probesCount, cellCounts.probesCount).CopyFrom(bakingCell.probePositions);
+                validity.GetSubArray(startCounts.probesCount, cellCounts.probesCount).CopyFrom(bakingCell.validity);
+                offsets.GetSubArray(startCounts.offsetsCount, cellCounts.offsetsCount).CopyFrom(bakingCell.offsetVectors);
+
+                startCounts.Add(cellCounts);
+            }
+
+            // Need to save here because the forced import below discards the changes.
+            EditorUtility.SetDirty(asset);
+            AssetDatabase.SaveAssets();
+
+            GetBlobFileNames(asset.GetSerializedFullPath(), out var cellDataFilename, out var cellSupportDataFilename);
+
+            unsafe
+            {
+                static long AlignRemainder16(long count) => count % 16L;
+
+                using (var fs = new System.IO.FileStream(cellDataFilename, System.IO.FileMode.Create, System.IO.FileAccess.Write))
+                {
+                    fs.Write(new ReadOnlySpan<byte>(bricks.GetUnsafeReadOnlyPtr(), bricks.Length * UnsafeUtility.SizeOf<Brick>()));
+                    fs.Write(new byte[AlignRemainder16(fs.Position)]);
+                    fs.Write(new ReadOnlySpan<byte>(probes.GetUnsafeReadOnlyPtr(), probes.Length * UnsafeUtility.SizeOf<Vector4>()));
+                }
+                using (var fs = new System.IO.FileStream(cellSupportDataFilename, System.IO.FileMode.Create, System.IO.FileAccess.Write))
+                {
+                    fs.Write(new ReadOnlySpan<byte>(positions.GetUnsafeReadOnlyPtr(), positions.Length * UnsafeUtility.SizeOf<Vector3>()));
```

Comment:
I have a funny feeling both position and validity will be needed as part of the main data soon :P But fair enough for now.